### PR TITLE
Feature: Slash Commands and Context Menu Commands

### LINKED ISSUE #94: The user should be able to pin/unpin messages through apps commands
https://discord.com/developers/docs/interactions/application-commands

### LINKED ISSUE #108: Args not recognized when space is not the delimiter
https://github.com/Section-IG/DataDrop/blob/1b0315f8748e381e6a241ba73980417eb7a3726b/src/events/messageCreate.ts#L75

The code splits on a space. However, if you are using other delimiters like newlines (e.g. calling the `announce`), then recognizing
(1) the command
(2) the args
will not work as we don't take account of the newline delimiter.

Example:
image
image

### PR DESCRIPTION
**Pull request by HunteRoi**
## Objective
Rework the command handling system to get rid of prefixes and archaic way of finding commands and arguments.

## Side-effects
- Rework of all commands behaviour for consistency
- Event handling system also got updated for consistency
- Fix for small warnings and errors (docker, db migrations, etc.)
- Linter moved from eslint to biomejs
- Fix an misconception where docker would never write to the bot_logs volume

Fixes #108, closes #94

### CODE REVIEW

File: src/commands/owner/eval.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        const { config } = client;
+        if (!config.ownerIds.includes(interaction.user.id)) {
+            await interaction.reply({
+                content:
+                    "❌ **Oups!** - Vous n'êtes pas autorisé à utiliser cette commande.",
+                ephemeral: true,
+            });
+            return;
+        }
+
```
Comment (HunteRoi) [inline src/commands/owner/eval.ts]:
```suggestion
        const { ownerIds } = client.config;
        if (!ownerIds.includes(interaction.user.id)) {
            await interaction.reply({
                content:
                    "❌ **Oups!** - Vous n'êtes pas autorisé à utiliser cette commande.",
                ephemeral: true,
            });
            return;
        }

```

File: src/commands/owner/reload.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        const { config } = client;
+        if (!config.ownerIds.includes(interaction.user.id)) {
+            await interaction.reply({
+                content:
+                    "❌ **Oups!** - Vous n'êtes pas autorisé à utiliser cette commande.",
+                ephemeral: true,
+            });
+            return;
+        }
+
```
Comment (HunteRoi) [inline src/commands/owner/reload.ts]:
```suggestion
        // double check sur l'identité juste pour la sécurité
        const { ownerIds } = client.config;
        if (!ownerIds.includes(interaction.user.id)) {
            await interaction.reply({
                content:
                    "❌ **Oups!** - Vous n'êtes pas autorisé à utiliser cette commande.",
                ephemeral: true,
            });
            return;
        }

```

File: src/commands/owner/restart.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        const { config } = client;
+        if (!config.ownerIds.includes(interaction.user.id)) {
+            await interaction.reply({
+                content:
+                    "❌ **Oups!** - Vous n'êtes pas autorisé à utiliser cette commande.",
+                ephemeral: true,
+            });
+            return;
+        }
+
```
Comment (HunteRoi) [inline src/commands/owner/restart.ts]:
```suggestion
        // double check sur l'identité juste pour la sécurité
        const { ownerIds } = client.config;
        if (!ownerIds.includes(interaction.user.id)) {
            await interaction.reply({
                content:
                    "❌ **Oups!** - Vous n'êtes pas autorisé à utiliser cette commande.",
                ephemeral: true,
            });
            return;
        }

```

File: src/models/Event.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import type { DatadropClient } from "../datadrop.js";
+
+export interface Event {
+    name: string;
```
Comment (HunteRoi) [inline src/models/Event.ts]: The event name could be type-secured to force specific values. That could also help to design an execute function that takes only specific params depending on the event's name.

File: src/services/CommandHandler.ts [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+            );
+            const replyOptions = {
+                content:
+                    "❌ **Oups!** - Une erreur est survenue en essayant cette commande. Reporte-le à un membre du Staff s'il te plaît!",
+                ephemeral: true,
+            };
+            if (interaction.replied) {
+                await interaction.followUp(replyOptions);
+            } else if (interaction.deferred) {
+                await interaction.editReply(replyOptions.content);
```
Comment (HunteRoi) [inline src/services/CommandHandler.ts]: Maybe first start with deferred then replied?

**Review by HunteRoi — COMMENTED** [review]
image

File: Dockerfile
Comment (DrakeXorn) [inline Dockerfile:1]: I don't really see the point of changing this but okay

File: scripts/deploy-commands.cjs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        //guildId // to deploy commands to a specific guild
+    });
+    client.login();
+});
+
+/* DELETE ALL DEPLOYED GLOBAL COMMANDS */
+// const rest = new REST().setToken(process.env.DISCORD_TOKEN);
+// rest.put(Routes.applicationCommands('703031563062870107'), { body: [] })
+//     .then(() => console.log('Successfully deleted all application commands.'))
+//     .catch(console.error);
```
Comment (DrakeXorn) [inline scripts/deploy-commands.cjs]: Why commenting instead of having a helper that does this kind of thing?

File: src/commands/others/pin.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+} from "discord.js";
+
+import type { DatadropClient } from "../../datadrop.js";
+import type { Command } from "../../models/Command.js";
+
+export default {
+    data: new ContextMenuCommandBuilder()
+        .setName("(Dés)Épingler le message")
+        .setType(ApplicationCommandType.Message)
+        // .setDescription("(Dés)épingle le message sélectionné dans le canal.")
```
Comment (DrakeXorn) [inline src/commands/others/pin.ts]: Wouldn't it be better to delete this line of code instead of commenting it?

File: yarn.lock
Comment (DrakeXorn) [inline yarn.lock:1]: Why deleting this?

Comment (HunteRoi) [reply yarn.lock:1]: It's an update of the dependencies, it's not deleted. I think GitHub is just messing with the state here.
image

File: src/commands/others/pin.ts (hunk 7 above)
Comment (HunteRoi) [reply src/commands/others/pin.ts]: You are right! Thanks for pointing it out :)
```suggestion
```

File: Dockerfile
Comment (HunteRoi) [reply Dockerfile:1]: Just to fix a few warnings from the docker CLI, but there's really no difference in regards of how the image is built anyways.

File: scripts/deploy-commands.cjs (hunk 6 above)
Comment (HunteRoi) [reply scripts/deploy-commands.cjs]: This file was a quick deploy script but can obviously be refactored. I will take a look at it, thanks!

**Review by DrakeXorn — APPROVED** [review]
Okay, now it’s ready :^)

**Pull request merged**